text/0000-unicode-string.md

```diff
@@ -0,0 +1,97 @@
+- Feature Name: unicode-string
+- Start Date: 2020-07-30
+- RFC PR: (leave this empty)
+- Pony Issue: (leave this empty)
+
+# Summary
+
+Change the builtin String class to present a sequence of unicode codepoints as 4 byte numbers (U32) instead of a sequence of bytes. All conversions from a sequence of bytes to a String, or from a String to a sequence of bytes will require specifying a decoding or encoding respectively. The default encoding will be UTF-8, and Strings will be represented internally as a sequence of UTF-8 encoded codepoints. Provide encoders/ decoders for UTF-8, UTF-16, UTF-32, ASCII and ISO-8859-1 as part of the stdlib. Change the character literal to represent a single unicode codepoint and produce a UTF-32 codepoint value.
```

Comment:
It is not 100% clear to me what the internal representation of String will be. Will it remain a pointer and a size only that the bytes pointed to will be actual `U32` codepoints without encoding? Or will the actual underlying data remain encoded? This RFC should also state why this approach was chosen and the other ones dicarded.

Reply:
I get the sense that by "present a sequence..." he may just mean that it implements `Seq[U32]` and would support those APIs and that as he's said below the data would still be represented internally as UTF-8.

Reply:
So to make sure I understand correctly, "hello" would be internally stored as five bytes: ```104, 101, 108, 108, 111```, but if we asked for third character it would return ```0,0,0,108``` (assuming endian)?

But, when you stray out of ASCII into codepoints, for example: 
```
"hӧle" => 104, **211, 167**, 108, 101
```
Would still have a length of 4, and the third character would be l?